language/testing-infra/diem-transactional-test-harness/src/diem_test_harness.rs

```diff
@@ -0,0 +1,395 @@
+    },
+    chain_id::ChainId,
+    transaction::{
+        Module as TransactionModule, RawTransaction, Script as TransactionScript,
+        ScriptFunction as TransactionScriptFunction, SignedTransaction, Transaction,
+        TransactionStatus,
+    },
+    vm_status::KeptVMStatus,
+};
+use diem_vm::DiemVM;
+use language_e2e_tests::data_store::{FakeDataStore, GENESIS_CHANGE_SET_FRESH};
+use move_binary_format::file_format::{CompiledModule, CompiledScript};
+use move_core_types::{
+    account_address::AccountAddress,
+    gas_schedule::{GasAlgebra, GasConstants},
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, ResourceKey, TypeTag},
+    move_resource::MoveStructType,
+    transaction_argument::{convert_txn_args, TransactionArgument},
+};
+use move_lang::{shared::verify_and_create_named_address_mapping, FullyCompiledProgram};
+use move_transactional_test_runner::{
+    framework::{run_test_impl, CompiledState, MoveTestAdapter},
+    tasks::{EmptyCommand, InitCommand, SyntaxChoice, TaskInput},
+    vm_test_harness::view_resource_in_move_storage,
+};
+use once_cell::sync::Lazy;
+use std::{
+    collections::{BTreeMap, HashMap},
+    path::Path,
+};
+use structopt::StructOpt;
+
+/// The Diem transaction test adapter.
+///
+/// This differs from the SimpleVMTestAdapter in a few ways to ensure that our tests mimics
+/// production settings:
+///   - It uses a Diem StateView as its storage backend
+///   - It executes transactions through DiemVM, instead of MoveVM directly
+struct DiemTestAdapter<'a> {
+    compiled_state: CompiledState<'a>,
+    storage: FakeDataStore,
+    default_syntax: SyntaxChoice,
+}
+
+/// Parameters *required* to create a Diem transaction.
+struct TransactionParameters {
+    pub sequence_number: u64,
+    pub max_gas_amount: u64,
+    pub gas_unit_price: u64,
+    pub gas_currency_code: String,
+    pub expiration_timestamp_secs: u64,
+}
+
+fn parse_ed25519_private_key(s: &str) -> Result<Ed25519PrivateKey> {
+    Ok(Ed25519PrivateKey::from_encoded_string(s)?)
+}
+
+/// Diem-specific arguments for the publish command.
+#[derive(StructOpt, Debug)]
+struct DiemPublishArgs {
+    #[structopt(short = "k", long = "private-key", parse(try_from_str = parse_ed25519_private_key))]
+    privkey: Ed25519PrivateKey,
+}
+
+/// Diem-specifc arguments for the run command,
+#[derive(StructOpt, Debug)]
+struct DiemRunArgs {
+    #[structopt(short = "k", long = "private-key", parse(try_from_str = parse_ed25519_private_key))]
+    privkey: Ed25519PrivateKey,
+}
+
+impl<'a> DiemTestAdapter<'a> {
+    /// Obtain a Rust representation of the account resource from storage, which is used to derive
+    /// a few default transaction parameters.
+    fn fetch_account_resource(&self, signer_addr: &AccountAddress) -> Result<AccountResource> {
+        let account_access_path = AccessPath::resource_access_path(ResourceKey::new(
+            *signer_addr,
+            AccountResource::struct_tag(),
+        ));
```

Comment:
These don't look they weren't that bad to implement. Glad we kicked `FakeExecutor`!